### Initial Analysis

In [4]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

##### A. Load the Foreign Gifts data.

In [5]:
foreign_gifts = pd.read_csv("ForeignGifts_edu.csv", low_memory=False)
print(foreign_gifts.shape)
foreign_gifts.head()

(28221, 10)


,ID,OPEID,Institution Name,City,State,Foreign Gift Received Date,Foreign Gift Amount,Gift Type,Country of Giftor,Giftor Name
0,1,102000,Jacksonville State University,Jacksonville,AL,43738,250000,Monetary Gift,CHINA,NaN
1,2,104700,Troy University,Troy,AL,43592,463657,Contract,CHINA,Confucius Institute Headquarters
2,3,105100,University of Alabama,Tuscaloosa,AL,43466,3649107,Contract,ENGLAND,Springer Nature Customer Service Ce
3,4,105100,University of Alabama,Tuscaloosa,AL,43472,1000,Contract,SAUDI ARABIA,Saudi Arabia Education Mission
4,5,105100,University of Alabama,Tuscaloosa,AL,43479,49476,Contract,SAUDI ARABIA,Saudi Arabia Education Mission


#####  B. Describe the differences between different classes of gift types.

In [ ]:
foreign_gifts.columns

Index(['ID', 'OPEID', 'Institution Name', 'City', 'State',
       'Foreign Gift Received Date', 'Foreign Gift Amount', 'Gift Type',
       'Country of Giftor', 'Giftor Name'],
      dtype='object')

In [7]:
foreign_gifts["Gift Type"].value_counts()

Gift Type
Contract         17274
Monetary Gift    10936
Real Estate         11
Name: count, dtype: int64

In [9]:
pd.options.display.float_format = "{:,.2f}".format
gift_type_summary = foreign_gifts.groupby("Gift Type")["Foreign Gift Amount"].describe()
gift_type_summary

,count,mean,std,min,25%,50%,75%,max
Gift Type,,,,,,,,
Contract,"17,274.00","625,794.51","3,768,216.87","-537,770.00","13,757.50","110,000.00","366,000.00","99,999,999.00"
Monetary Gift,"10,936.00","528,150.44","2,086,487.12",1.00,250.00,"50,000.00","400,000.00","48,842,186.00"
Real Estate,11.00,"1,335,265.45","1,412,070.57","39,456.00","370,574.50","667,555.00","2,121,736.00","4,312,000.00"




  The results show several notable differences between the different classes of gift types. Contracts are the most common with 17,274 records, followed by monetary gifts with 10,936. What caught my attention was the very low number of real estate gifts, with only 11 records. Despite having the smallest count, real estate has the largest mean at about $1.34 million. Contracts and monetary gifts have lower and somewhat similar means, with contracts averaging $625,794.51 and monetary gifts averaging $528,150.44.

  Another thing to point out is the large standard deviation across all three gift types, which shows that the gift amounts vary considerably. We can also see a large difference between the mean and median. Contracts have a median of $110,000, monetary gifts have a median of $50,000, and real estate gifts have a median of $667,555. Since the means are much higher than the medians, we can infer that some very large gift amounts are pulling the means upward.


##### C. Answer the following questions:


a. Which country gives the most money in total?

b. Which country initiates the most gifts by count?

c. Which country gives the largest gifts on average?

d. Which institution receives the most money in total?

e. Which institution receives the greatest number of gifts by count?

f. What is the average amount each university receives? Median amount? If they are different, why?

g. What is the largest flow of a single country to a single institution?

h. Which are the top giftors to US academic institutions?

  a. Which country gives the most money in total?

  Qatar is the country that gives the most money in total, with a total gift amount of $2.71 billion.

In [10]:
country_total_gifts = foreign_gifts.groupby("Country of Giftor")["Foreign Gift Amount"].sum()
country_total_gifts

Country of Giftor
AFGHANISTAN                  2472377
AMERICAN SAMOA               2338347
ANGOLA                        932390
ANTIGUA                      3435545
ARGENTINA                    8464084
                              ...   
VIRGIN ISLANDS (BRITISH)    13470572
WALES                        3901656
YEMEN                         586537
ZAMBIA                      17516086
ZIMBABWE                      358301
Name: Foreign Gift Amount, Length: 155, dtype: int64

In [14]:
country_total_gifts.nlargest(3)

Country of Giftor
QATAR      2706240869
ENGLAND    1464906771
CHINA      1237952112
Name: Foreign Gift Amount, dtype: int64

b. Which country initiates the most gifts by count?

England is the country that initiates the most gifts, with a total of 3,655 gifts.

In [13]:
country_gift_counts = foreign_gifts["Country of Giftor"].value_counts()
country_gift_counts.nlargest(3)

Country of Giftor
ENGLAND    3655
CHINA      2461
CANADA     2344
Name: count, dtype: int64

c. Which country gives the largest gifts on average?

Bermuda gives the largest gifts on average, with an average gift amount of approximately $7,688,837.37.

In [15]:
country_average_gifts = foreign_gifts.groupby("Country of Giftor")["Foreign Gift Amount"].mean()
country_average_gifts.nlargest(3)

Country of Giftor
BERMUDA   7,688,837.37
ECUADOR   3,985,140.29
QATAR     3,905,109.48
Name: Foreign Gift Amount, dtype: float64

d. Which institution receives the most money in total?

Carnegie Mellon University receives the most money in total, with approximately $1.48 billion in foreign gifts.

In [16]:
institution_total_gifts = foreign_gifts.groupby("Institution Name")["Foreign Gift Amount"].sum()
institution_total_gifts.nlargest(3)

Institution Name
Carnegie Mellon University    1477922504
Cornell University            1289937761
Harvard University             954803610
Name: Foreign Gift Amount, dtype: int64

e. Which institution receives the greatest number of gifts by count?

The University of California, Los Angeles receives the greatest number of gifts, with a total of 3,916 gift records


In [17]:
institution_gift_counts = foreign_gifts["Institution Name"].value_counts()
institution_gift_counts.nlargest(3)

Institution Name
University of California, Los Angeles          3916
California Institute of Technology             2502
Columbia University in the City of New York    1127
Name: count, dtype: int64

f. What is the average amount each university receives? Median amount? If they are different, why?

The mean is approximately $52.2 million, while the median is approximately $6.49 million. As we can see, the mean and median differ by a lot, suggesting that some very large values may be pulling the mean upward.

In [19]:
average_institution_gifts = institution_total_gifts.mean()
median_institution_gifts = institution_total_gifts.median()

print(f"Average: ${average_institution_gifts:,.2f}")
print(f"Median: ${median_institution_gifts:,.2f}")

Average: $52,202,878.97
Median: $6,486,791.50


g. What is the largest flow of a single country to a single institution?

The largest flow from a single country to a single institution is from Qatar to Cornell University, with a total of approximately $1.02 billion.

In [20]:
country_institution_total = foreign_gifts.groupby(
    ["Country of Giftor", "Institution Name"]
)["Foreign Gift Amount"].sum()

country_institution_total.nlargest(3)

Country of Giftor  Institution Name          
QATAR              Cornell University            1018473315
BERMUDA            Carnegie Mellon University     750000000
QATAR              Texas A&M University           504099350
Name: Foreign Gift Amount, dtype: int64